open-sphere-base/mantle/src/main/java/io/opensphere/mantle/icon/FontIcons.java

```diff
@@ -0,0 +1,297 @@
+        }
+        return ourIconNameToUrlMap;
+    }
+
+    /**
+     * Initializes the system properties map. We only care about the runtime
+     * path value.
+     */
+    private static void buildPropertiesMap()
+    {
+        Properties p = System.getProperties();
+        String pathProperty = "opensphere.path.runtime";
+        String pathValue = StringUtilities.expandProperties(p.getProperty(pathProperty), p);
+
+        ourSystemPropertiesMap.put(pathProperty, pathValue);
+    }
+
+    /**
+     * Initialize the default icons.
+     *
+     * @param mantleToolbox Reference to the MantleToolbox
+     * @return the map
+     */
+    private static Map<String, URL> loadIcons(MantleToolbox mantleToolbox)
+    {
+        List<IconRecord> records = new LinkedList<>(getIconsFromRegistry(mantleToolbox));
+
+        Map<String, Integer> nameToIdMap = records.stream()
+                .collect(Collectors.toMap(r -> r.getName(), r -> Integer.valueOf(r.getId()), (v1, v2) -> v2));
+
+        removeOldRecords(mantleToolbox, records);
+
+        if (records.isEmpty())
+        {
+            records = readIconsFromFile(mantleToolbox, nameToIdMap);
+        }
+
+        return createMap(records);
+    }
+
+    /**
+     * Gets the KML icon records from the icon registry.
+     *
+     * @param mantleToolbox the mantle toolbox
+     * @return the icon records
+     */
+    private static List<IconRecord> getIconsFromRegistry(MantleToolbox mantleToolbox)
+    {
+        return mantleToolbox.getIconRegistry().getIconRecords(r -> "Font Icons".equals(r.getCollectionName()));
+    }
+
+    /**
+     * Removes old records from the list and the icon registry.
+     *
+     * @param mantleToolbox the mantle toolbox
+     * @param records the icon records
+     */
+    private static void removeOldRecords(MantleToolbox mantleToolbox, Collection<IconRecord> records)
+    {
+        if (!records.isEmpty())
+        {
+            TIntList removedIds = new TIntArrayList();
+            for (Iterator<IconRecord> iter = records.iterator(); iter.hasNext();)
+            {
+                IconRecord record = iter.next();
+
+                iter.remove();
+                removedIds.add(record.getId());
+            }
+
+            if (!removedIds.isEmpty())
+            {
+                mantleToolbox.getIconRegistry().removeIcons(removedIds, FontIcons.class);
+            }
+        }
+    }
+
+    /**
+     * Reads icons from the file, into the icon registry.
+     *
+     * @param mantleToolbox the mantle toolbox
+     * @param nameToIdMap map of icon name to icon ID (to keep IDs consistent)
+     * @return the icon records
+     */
+    private static List<IconRecord> readIconsFromFile(MantleToolbox mantleToolbox, Map<String, Integer> nameToIdMap)
+    {
+        List<IconProvider> iconProviders;
+
+        Stream<IconProvider> fontawesomeStream = Arrays.stream(AwesomeIcon.values()).map(FontIcons::getResource)
```

Comment:
Would it make sense to move code like this and the logic in the getResource() methods below into the individial Icon classes?  I think it would be better encapsulated and make adding more icon sets in the future a little simpler as this file wouldn't need to be changed.

Reply:
Ideally, I think in the future this class would be replaced by a different method of setting up icons. The way I have in mind, there would be an alternative menu in the IconChooser that allows a user to change icon features (color, default size) before saving the custom image to the registry, as opposed to pre-generating everything like in this class.

That would take a bit more work, though.